src/xmpp/session.c

```diff
@@ -286,6 +286,36 @@ session_get_account_name(void)
     return saved_account.name;
 }
 
+static int
+_receive_mood(xmpp_stanza_t* const stanza, void* const userdata)
+{
+    const char* from = xmpp_stanza_get_from(stanza);
+    xmpp_stanza_t* event = xmpp_stanza_get_child_by_name_and_ns(stanza, STANZA_NAME_EVENT, STANZA_NS_PUBSUB_EVENT);
+    if (event) {
+        xmpp_stanza_t* items = xmpp_stanza_get_child_by_name(event, STANZA_NAME_ITEMS);
+        if (items) {
+            xmpp_stanza_t* item = xmpp_stanza_get_child_by_name(items, STANZA_NAME_ITEM);
+            if (item) {
+                xmpp_stanza_t* mood = xmpp_stanza_get_child_by_name_and_ns(item, STANZA_NAME_MOOD, STANZA_NS_MOOD);
+                if (mood) {
+                    xmpp_stanza_t* c = xmpp_stanza_get_children(mood);
+                    if (c) {
+                        const char* m = xmpp_stanza_get_name(c);
+                        xmpp_stanza_t* t = xmpp_stanza_get_child_by_name(mood, STANZA_NAME_TEXT);
+                        if (t) {
+                            const char* text = xmpp_stanza_get_text(t);
+                            cons_show("Mood from %s %s (%s)", from, m, text);
+                        } else {
+                            cons_show("Mood from %s %s", from, m);
+                        }
+                    }
+                }
+            }
+        }
+    }
```

Comment:
when I saw this code I came up with strophe/libstrophe#187

Do you think this might help?

Reply:
Yes, I like that! Thanks!

Reply:
It's merged, now we only have to wait for a new release :)